From 2e3bec68b5a652281bb29f4fd0bf42f7045291b1 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Thu, 4 Jul 2019 22:12:02 -0700
Subject: [PATCH 0633/1851] v4l: xilinx: sdirxss: Make structures constant

commit d94be8f37a027a895bc38ac71338b8d99466af3b from
https://github.com/Xilinx/linux-xlnx.git

Mark v4l2 controls and subdev operation structures as constants.
Replace strlcpy with strscpy.
This is suggested while upstreaming the driver.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-sdirxss.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-sdirxss.c b/drivers/media/platform/xilinx/xilinx-sdirxss.c
index 1ee54ce28d22..1d3b2fa97265 100644
--- a/drivers/media/platform/xilinx/xilinx-sdirxss.c
+++ b/drivers/media/platform/xilinx/xilinx-sdirxss.c
@@ -1395,7 +1395,7 @@ static const struct v4l2_ctrl_ops xsdirxss_ctrl_ops = {
 	.s_ctrl	= xsdirxss_s_ctrl
 };
 
-static struct v4l2_ctrl_config xsdirxss_edh_ctrls[] = {
+static const struct v4l2_ctrl_config xsdirxss_edh_ctrls[] = {
 	{
 		.ops	= &xsdirxss_ctrl_ops,
 		.id	= V4L2_CID_XILINX_SDIRX_EDH_ERRCNT_ENABLE,
@@ -1427,7 +1427,7 @@ static struct v4l2_ctrl_config xsdirxss_edh_ctrls[] = {
 	}
 };
 
-static struct v4l2_ctrl_config xsdirxss_ctrls[] = {
+static const struct v4l2_ctrl_config xsdirxss_ctrls[] = {
 	{
 		.ops	= &xsdirxss_ctrl_ops,
 		.id	= V4L2_CID_XILINX_SDIRX_FRAMER,
@@ -1512,17 +1512,17 @@ static const struct v4l2_subdev_core_ops xsdirxss_core_ops = {
 	.unsubscribe_event = xsdirxss_unsubscribe_event
 };
 
-static struct v4l2_subdev_video_ops xsdirxss_video_ops = {
+static const struct v4l2_subdev_video_ops xsdirxss_video_ops = {
 	.g_frame_interval = xsdirxss_g_frame_interval,
 	.s_stream = xsdirxss_s_stream
 };
 
-static struct v4l2_subdev_pad_ops xsdirxss_pad_ops = {
+static const struct v4l2_subdev_pad_ops xsdirxss_pad_ops = {
 	.get_fmt = xsdirxss_get_format,
 	.set_fmt = xsdirxss_set_format,
 };
 
-static struct v4l2_subdev_ops xsdirxss_ops = {
+static const struct v4l2_subdev_ops xsdirxss_ops = {
 	.core = &xsdirxss_core_ops,
 	.video = &xsdirxss_video_ops,
 	.pad = &xsdirxss_pad_ops
@@ -1720,7 +1720,7 @@ static int xsdirxss_probe(struct platform_device *pdev)
 
 	subdev->dev = &pdev->dev;
 	subdev->internal_ops = &xsdirxss_internal_ops;
-	strlcpy(subdev->name, dev_name(&pdev->dev), sizeof(subdev->name));
+	strscpy(subdev->name, dev_name(&pdev->dev), sizeof(subdev->name));
 
 	subdev->flags |= V4L2_SUBDEV_FL_HAS_EVENTS | V4L2_SUBDEV_FL_HAS_DEVNODE;
 
-- 
2.31.1

